### PR TITLE
Clarify suggestion for E0013

### LINKED ISSUE #68038: E0013 confusing diagnostic
```rust
static FOO: u8 = 0;
const FOO_REF: &u8 = &FOO;
```

gives 
```rust
error[E0013]: constants cannot refer to statics, use a constant instead
 --> src/lib.rs:3:22
  |
3 | const FOO_REF: &u8 = &FOO;
  |                      ^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0013`.
error: could not compile `playground`.
```

The wording `constants cannot refer to statics, use a constant instead` is confusing

### PR DESCRIPTION
**Pull request by varkor**
Fixes https://github.com/rust-lang/rust/issues/68038.

### CODE REVIEW

File: src/librustc_mir/transform/check_consts/ops.rs [hunk 1]
```diff
@@ -353,16 +353,16 @@ impl NonConstOp for StaticAccess {
             item.tcx.sess,
             span,
             E0013,
-            "{}s cannot refer to statics, use \
-                                        a constant instead",
+            "{}s cannot refer to statics",
             item.const_kind()
         );
+        err.help("consider extracting the value of the static to a const, and referring to that");
```
Comment (Centril) [inline src/librustc_mir/transform/check_consts/ops.rs]:
```suggestion
        err.help("consider extracting the value of the `static` to a `const`, and referring to that");
```

File: src/librustc_mir/transform/check_consts/ops.rs [hunk 2]
```diff
@@ -353,16 +353,16 @@ impl NonConstOp for StaticAccess {
...
         if item.tcx.sess.teach(&err.get_code().unwrap()) {
             err.note(
                 "Static and const variables can refer to other const variables. \
```
Comment (Centril) [inline src/librustc_mir/transform/check_consts/ops.rs]:
```suggestion
                "`static` and `const` variables can refer to other `const` variables. \
```

File: src/librustc_mir/transform/check_consts/ops.rs [hunk 3]
```diff
@@ -353,16 +353,16 @@ impl NonConstOp for StaticAccess {
...
                     But a const variable cannot refer to a static variable.",
             );
-            err.help("To fix this, the value can be extracted as a const and then used.");
+            err.help("To fix this, the value can be extracted to a const and then used.");
```
Comment (Centril) [inline src/librustc_mir/transform/check_consts/ops.rs]:
```suggestion
            err.help("To fix this, the value can be extracted to a `const` and then used.");
```

File: src/librustc_mir/transform/check_consts/ops.rs [hunk 4]
```diff
@@ -353,16 +353,16 @@ impl NonConstOp for StaticAccess {
...
-            "{}s cannot refer to statics, use \
-                                        a constant instead",
+            "{}s cannot refer to statics",
             item.const_kind()
         );
+        err.help("consider extracting the value of the static to a const, and referring to that");
         if item.tcx.sess.teach(&err.get_code().unwrap()) {
             err.note(
                 "Static and const variables can refer to other const variables. \
                     But a const variable cannot refer to a static variable.",
```
Comment (Centril) [inline src/librustc_mir/transform/check_consts/ops.rs]:
```suggestion
                    a `const` variable cannot however refer to a static variable.",
```
(+ split this sentence into a second `.note("...")`)

**Review by Centril — COMMENTED** [review]
r=me with comments addressed.

File: src/librustc_mir/transform/check_consts/ops.rs (hunk 4 above)
Comment (varkor) [reply src/librustc_mir/transform/check_consts/ops.rs]: There seems to be a different style for `--teach` diagnostics, in which multi-line notes are acceptable.

Comment (Centril) [reply src/librustc_mir/transform/check_consts/ops.rs]: ah 👍

**Pull request merged**